### PR TITLE
Create mock IpmiCommandLib under IpmiFeaturePkg

### PR DESCRIPTION
**Pull request by TsunFeng**
## Description

Create mock IpmiCommandLib under IpmiFeaturePkg

For details on how to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Unit tests component can call these mock functions success

## Integration Instructions

N/A

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

File: IpmiFeaturePkg/Test/Mock/Include/GoogleTest/Library/MockIpmiCommandLib.h [hunk 1]
```diff
@@ -0,0 +1,345 @@
+/** @file MockIpmiCommandLib.h
+  Google Test mocks for IpmiCommandLib
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline IpmiFeaturePkg/Test/Mock/Include/GoogleTest/Library/MockIpmiCommandLib.h:5]: This is an open source repo, this needs to be 

  Copyright (c) Microsoft Corporation
  SPDX-License-Identifier: BSD-2-Clause-Patent

File: IpmiFeaturePkg/Test/Mock/Library/GoogleTest/MockIpmiCommandLib/MockIpmiCommandLib.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/** @file MockIpmiCommandLib.cpp
+  Google Test mocks for IpmiCommandLib
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline IpmiFeaturePkg/Test/Mock/Library/GoogleTest/MockIpmiCommandLib/MockIpmiCommandLib.cpp:5]: Same here:

  Copyright (c) Microsoft Corporation.
  SPDX-License-Identifier: BSD-2-Clause-Patent

File: IpmiFeaturePkg/Test/Mock/Library/GoogleTest/MockIpmiCommandLib/MockIpmiCommandLib.inf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+## @file MockIpmiCommandLib.inf
+# Google Test mocks for IpmiCommandLib
+#
+# Copyright (c) Microsoft Corporation.
+# Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (apop5) [inline IpmiFeaturePkg/Test/Mock/Library/GoogleTest/MockIpmiCommandLib/MockIpmiCommandLib.inf:5]: And here

  Copyright (c) Microsoft Corporation.
  SPDX-License-Identifier: BSD-2-Clause-Patent

File: IpmiFeaturePkg/Test/IpmiFeaturePkgHostTest.dsc [hunk 4]
```diff
@@ -64,4 +64,5 @@
   #
   # Build HOST_APPLICATION Libraries With GoogleTest
   #
-  IpmiFeaturePkg/Test/Mock/Library/GoogleTest/MockIpmiSelLib/MockIpmiSelLib.inf
```
Comment (VivianNK) [inline IpmiFeaturePkg/Test/IpmiFeaturePkgHostTest.dsc:67]: Please remove whitespace change

**Review by VivianNK — APPROVED** [review]

**Pull request merged**